utils/src/main/java/software/amazon/awssdk/utils/async/SimplePublisher.java

```diff
@@ -72,27 +69,19 @@
      * <p>All logic within this publisher is represented using events in this queue. This ensures proper ordering of events
      * processing and simplified reasoning about thread safety.
      */
-    private final Queue<QueueEntry<T>> eventQueue = new ConcurrentLinkedQueue<>();
+    private final Queue<QueueEntry<T>> standardPriorityQueue = new ConcurrentLinkedQueue<>();
+    private final Queue<QueueEntry<T>> highPriorityQueue = new ConcurrentLinkedQueue<>();
 
     /**
-     * When processing the {@link #eventQueue}, these are the entries that should be skipped (and failed). This is used to
-     * safely drain the queue when there are urgent events needing processing, like a {@link Subscription#cancel()}.
-     */
-    private final Set<QueueEntry.Type> entryTypesToFail = new CopyOnWriteArraySet<>();
-
-    /**
-     * Whether the {@link #eventQueue} is currently being processed. Only one thread may read events from the queue at a time.
+     * Whether the {@link #standardPriorityQueue} is currently being processed. Only one thread may read events from the
```

Comment:
Nit: should it be `Whether the {@link #standardPriorityQueue} or {@link #highPriorityQueue} is currently being processed.`?